presto-main/src/main/java/com/facebook/presto/server/security/Authorizer.java

```diff
@@ -0,0 +1,21 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.server.security;
+
+import javax.servlet.http.HttpServletRequest;
+
+public interface Authorizer
+{
+    boolean authorize(HttpServletRequest request) throws AuthorizationException;
```

Comment:
nit: `boolean` does not make sense. 

Since this is an authorization interface, returning an exception does not seem like the right way. I would prefer that we return something like `AuthorizationResult` which has 2 fields - `boolean allowed` and `Optional<String> reason`.

Reply:
+1